packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmInstantImpl.kt

```diff
@@ -25,3 +29,13 @@ public data class RealmInstantImpl(override val seconds: Long, override val nano
         return "RealmInstant(epochSeconds=$epochSeconds, nanosecondsOfSecond=$nanosecondsOfSecond)"
     }
 }
+
+internal fun RealmInstant.toDuration(): Duration {
+    return epochSeconds.seconds + nanosecondsOfSecond.nanoseconds
+}
+
+internal fun Duration.toRealmInstant(): RealmInstant {
+    val seconds = this.inWholeSeconds.seconds
+    val nanos = (this - seconds)
```

Comment:
This looks a bit suspicious to me? Why is the end result of this nanoseconds?

Reply:
Ah yeah, seconds is also a `Duration` 🙈 ...it would help if the types where visible here.